src/Avalonia.Controls/Utils/StringUtils.cs

```diff
@@ -85,12 +85,16 @@ public static bool IsEndOfWord(string text, int index)
             // preceeded by lwsp.
             if (index > 0)
             {
-                var nextCodePoint = new Codepoint(text[index + 1]);
-
-                if (nextCodePoint.IsBreakChar)
+                if (index + 1 < text.Length)
                 {
-                    return true;
+                    var nextCodePoint = new Codepoint(text[index + 1]);
+
+                    if (nextCodePoint.IsBreakChar)
+                    {
+                        return true;
+                    }
                 }
+                return true;
```

Comment:
That doesn't seem right, basically any white space char except the one at index 0 will get here, and never get to the rest of the method. Is an `else` missing?

That said, after reviewing the rest of the method, it always return `true`, since a white space can never match the letter/symbol/punctuation categories below. The whole method can be replaced with `return index >= text.Length || new Codepoint(text[index]).IsWhiteSpace;`, probably not the intended behavior. Note that this was the case before your PR so you're not responsible for that, just thinking that we should probably fix that later and add some matching unit tests.